monkey/monkey_island/cc/ui/src/components/ui-components/AdvancedMultiSelect.js

```diff
@@ -1,121 +1,118 @@
-import React, {useState} from 'react';
+import React from 'react';
+import {Form} from 'react-bootstrap';
 
-import {Card, Button, Form} from 'react-bootstrap';
-import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
-import {faCheckSquare} from '@fortawesome/free-solid-svg-icons';
-import {faSquare} from '@fortawesome/free-regular-svg-icons';
 import {cloneDeep} from 'lodash';
 
 import {getComponentHeight} from './utils/HeightCalculator';
-import {resolveObjectPath} from './utils/ObjectPathResolver';
 import InfoPane from './InfoPane';
+import {MasterCheckbox, MasterCheckboxState} from './MasterCheckbox';
+import ChildCheckbox from './ChildCheckbox';
+import {getFullDefinitionByKey, getDefaultPaneParams} from './JsonSchemaHelpers.js';
 
+class AdvancedMultiSelect extends React.Component {
+  constructor(props) {
+    super(props)
 
-function getSelectValuesAfterClick(valueArray, clickedValue) {
-  if (valueArray.includes(clickedValue)) {
-    return valueArray.filter((e) => {
-      return e !== clickedValue;
-    });
-  } else {
-    valueArray.push(clickedValue);
-    return valueArray;
+    this.enumOptions = props.options.enumOptions;
+
+    this.state = {
+      masterCheckboxState: this.getMasterCheckboxState(props.value),
+      infoPaneParams: getDefaultPaneParams(props.schema.items.$ref, props.registry)
+    };
+
+    this.onMasterCheckboxClick = this.onMasterCheckboxClick.bind(this);
+    this.onChildCheckboxClick = this.onChildCheckboxClick.bind(this);
+    this.setPaneInfo = this.setPaneInfo.bind(this, props.schema.items.$ref, props.registry);
   }
-}
 
-function onMasterCheckboxClick(checkboxValue, defaultArray, onChangeFnc) {
-  if (checkboxValue) {
-    onChangeFnc([]);
-  } else {
-    onChangeFnc(defaultArray);
+  onMasterCheckboxClick() {
+    if (this.state.masterCheckboxState === MasterCheckboxState.ALL) {
+      var newValues = [];
+    } else {
+      newValues = this.enumOptions.map(({value}) => value);
+    }
+
+    this.props.onChange(newValues);
+    this.setMasterCheckboxState(newValues);
   }
-}
 
-// Definitions passed to components only contains value and label,
-// custom fields like "info" or "links" must be pulled from registry object using this function
-function getFullDefinitionsFromRegistry(refString, registry) {
-  return getObjectFromRegistryByRef(refString, registry).anyOf;
-}
+  onChildCheckboxClick(value) {
+    let selectValues = this.getSelectValuesAfterClick(value)
```

Comment:
```suggestion
    let selectValues = this.getSelectValuesAfterClick(value);
```

Reply:
Weird! I'm surprised this compiled/ran. TIL "automatic semicolon insertion": http://es5.github.io/#x7.9